src/sparseml/pytorch/image_classification/export.py

```diff
@@ -260,6 +261,19 @@ class ExportArgs:
     )
 
     def __post_init__(self):
+        if self.arch_key is None:
+            assert self.checkpoint_path, (
+                "Must provide a checkpoint path if " "no arch_key is provided"
+            )
+
+            checkpoint = torch.load(self.checkpoint_path)
```

Comment:
let's avoid doing the load and assert here since we only want to load from torch once (it can take a while to read in the data for larger models).

Reply:
refer to [this comment](https://github.com/neuralmagic/sparseml/pull/582#issuecomment-1059006841)